keyboards/doppelganger/doppelganger.c

```diff
@@ -0,0 +1,67 @@
+/* Copyright 2019 Yiancar-Designs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "doppelganger.h"
+
+// Optional override functions below.
+// You can leave any or all of these undefined.
+// These are only required if you want to perform custom actions.
+
+/*
+
+void matrix_init_kb(void) {
+  // put your keyboard start-up code here
+  // runs once when the firmware starts up
+
+  matrix_init_user();
+}
+
+void matrix_scan_kb(void) {
+  // put your looping keyboard code here
+  // runs every cycle (a lot)
+
+  matrix_scan_user();
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+  // put your per-action keyboard code here
+  // runs for every action, just before processing by the firmware
+
+  return process_record_user(keycode, record);
+}
+
+*/
+
+bool led_update_kb(led_t led_state) {
+  bool res = led_update_user(led_state);
+  if(res) {
+    // writePin sets the pin high for 1 and low for 0.
+    // In this example the pins are inverted, setting
+    // it low/0 turns it on, and high/1 turns the LED off.
+    // This behavior depends on whether the LED is between the pin
+    // and VCC or the pin and GND.
+    writePin(C6, !led_state.caps_lock);
+  }
+  return res;
+}
+
+__attribute__((weak)) layer_state_t layer_state_set_user(layer_state_t state) {
+  if( IS_LAYER_ON(1) ){
+    writePinLow(B0);
+  } else {
+    writePinHigh(B0);
+  }
```

Comment:
Would be better as: 
```c
    writePin(B0, (state & (1UL << 1)))
```

The `IS_LAYER_ON(1)` option calls `layer_state`, not `state`, so it may cause issues with timing?